bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/HikvisionHandler.java

```diff
@@ -64,18 +64,13 @@ public HikvisionHandler(ThingHandler handler, int nvrChannel) {
     // This handles the incoming http replies back from the camera.
     @Override
     public void channelRead(@Nullable ChannelHandlerContext ctx, @Nullable Object msg) throws Exception {
-        if (msg == null || ctx == null) {
+        if (msg == null || ctx == null || msg.toString().isEmpty()) {
             return;
         }
-        String content = "";
-        int debounce = 3;
         try {
-            content = msg.toString();
-            if (content.isEmpty()) {
-                return;
-            }
+            int debounce = 3;
+            String content = msg.toString();
             logger.trace("HTTP Result back from camera is \t:{}:", content);
-
```

Comment:
`.toString()` might be an expensive operation, so calling it twice when you only need to call it once is a bad practice. The way it was before only called it once. What were you trying to address here with this change?

Reply:
FIXED: I looked into it and what I was trying to address does not happen anymore. I have removed it back to only calling it once, thanks for pointing it out.